lib/util/common.js

```diff
@@ -210,11 +213,76 @@ function resolveSwaggerFunction (opts, cache, routes, Ref, done) {
   }
 }
 
+function localSchemaRefToAbs (schema, fullPath, pathFromLastId) {
+  if (schema.$id) {
+    if (!fullPath) {
+      fullPath = schema.$id + '#'
+    }
+    // start from the last obj with $id
+    pathFromLastId = ''
+  }
+
+  Object.keys(schema).forEach(key => {
+    if (key === '$ref' && schema[key] && schema[key].startsWith('#')) {
+      // #/something/... -> something/...
+      const woLocalSymbol = schema[key].substring(2)
+
+      // where ObjectA is last obj with #id
+      // where pathFromLastId is the path from ObjectA to the current object
+
+      // root#/properties/ObjectA/properties -> root#/properties/ObjectA
+      // todo: cov fail ???
+      /* istanbul ignore else */
+      if (pathFromLastId) {
+        const pathIdxForReplace = fullPath.lastIndexOf(pathFromLastId)
+        fullPath = fullPath.substring(0, pathIdxForReplace)
+      }
+
+      // handle cases like '#' or '#/something'
+      const absPath = woLocalSymbol ? fullPath + '/' + woLocalSymbol : fullPath
+      schema[key] = absPath
+    } else if (schema[key] && typeof schema[key] === 'object') {
+      const isRefObj = schema[key].$ref
+      schema[key] = localSchemaRefToAbs(schema[key],
+        // don't extend paths if we are going to $ref object
+        isRefObj ? fullPath : fullPath + '/' + key,
+        isRefObj ? pathFromLastId : pathFromLastId + '/' + key
+      )
+    }
+  })
+
+  return schema
+}
+
+function patchDefinitionsKeywordInSchema (schema) {
+  if (!schema) return schema
+
+  Object.keys(schema).forEach(key => {
+    if (key === '$ref' && schema[key]) {
+      schema[key] = schema[key].split('definitions').join('properties')
+    } else if (schema[key] && typeof schema[key] === 'object') {
+      if (key === 'definitions') {
+        schema.properties = {
```

Comment:
Why do you move schema from the `definitions` to the `properties`? You shouldn't create new properties.